agent/custom/general_navigation_action.py

```
@@ -624,306 +354,425 @@ def _iou(a, b):
+def _center_roi(img, cx, cy, max_w, max_h, pad=ROI_PAD):
+    """以 (cx,cy) 为中心, 按地图内最大模板尺寸+四周pad 画 ROI(BGR, clamp 到图内)
+    解决 YOLO 框不完整(局部框只覆盖名称条一部分)导致匹配低分的问题:
+    框中心基本可靠, ROI 保证完整包住名称条, 使多尺度模板匹配稳定 0.9+"""
+    H, W = img.shape[:2]
+    w, h = max_w + pad * 2, max_h + pad * 2
+    x0 = max(0, min(int(cx - w / 2), W - w))
+    y0 = max(0, min(int(cy - h / 2), H - h))
+    return img[y0:y0 + h, x0:x0 + w]
+
+
+def _mt_best(crop_bgr, tpl_bgr):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_mt_best`

(ANN202)